pkg/backends/backend.go

```diff
@@ -2,75 +2,74 @@ package backends
 
 import (
 	"context"
-	"fmt"
-	"time"
 
 	"go.uber.org/zap"
 	"k8s.io/client-go/kubernetes"
 	coreListersV1 "k8s.io/client-go/listers/core/v1"
 
-	"github.com/hellofresh/kangal/pkg/backends/fake"
-	"github.com/hellofresh/kangal/pkg/backends/jmeter"
-	"github.com/hellofresh/kangal/pkg/backends/locust"
 	loadTestV1 "github.com/hellofresh/kangal/pkg/kubernetes/apis/loadtest/v1"
 	clientSetV "github.com/hellofresh/kangal/pkg/kubernetes/generated/clientset/versioned"
 )
 
-// LoadTestType defines the methods that a loadtest type needs to implement
-// for the controller to be able to run it
-type LoadTestType interface {
-	// CheckOrCreateResources check for resources or create the needed resources for the loadtest type
-	CheckOrCreateResources(ctx context.Context) error
+// Backend is basic methods a backend must implement
+type Backend interface {
+	// Type must return backend unique type string, called by both Proxy and Controller
+	Type() loadTestV1.LoadTestType
 
-	// CheckOrUpdateLoadTestStatus check current LoadTest progress
-	CheckOrUpdateStatus(ctx context.Context) error
+	// TransformLoadTestSpec should validate and transform LoadTestSpec, called by Proxy
+	TransformLoadTestSpec(spec *loadTestV1.LoadTestSpec) error
```

Comment:
What is the meaning of `TransformLoadTestSpec`? 
Does it mean that this method transforms spec to backend?

Reply:
Means the backend has a chance to transform/validate the `LoadTestSpec` to ensure it matches his conditions